arithmetization/src/main/java/net/consensys/linea/zktracer/module/ext/ExtOperation.java

```diff
@@ -59,39 +61,41 @@ public class ExtOperation {
    */
   @Override
   public int hashCode() {
-    return Objects.hash(this.opCode, this.arg1, this.arg2, this.arg3);
-  }
-
-  public ExtOperation(OpCodeData opCodeData, Bytes32 arg1, Bytes32 arg2, Bytes32 arg3) {
-    this(opCodeData.mnemonic(), arg1, arg2, arg3);
+    return Objects.hash(this.opCode, this.rawArg1, this.rawArg2, this.rawArg3);
   }
 
   public ExtOperation(OpCode opCode, Bytes32 arg1, Bytes32 arg2, Bytes32 arg3) {
     this.opCode = opCode;
-    this.arg1 = BaseBytes.fromBytes32(arg1);
-    this.arg2 = BaseBytes.fromBytes32(arg2);
-    this.arg3 = BaseBytes.fromBytes32(arg3);
-    this.aBytes = BaseTheta.fromBytes32(arg1);
-    this.bBytes = BaseTheta.fromBytes32(arg2);
-    this.cBytes = BaseTheta.fromBytes32(arg3);
+    this.rawArg1 = arg1.copy();
+    this.rawArg2 = arg2.copy();
+    this.rawArg3 = arg3.copy();
+    this.oli = isOneLineInstruction();
+  }
+
+  public void setup() {
```

Comment:
So we do the computations required by EXT not when defining an `ExtOperation` but in the tracing loop at the end in the `commit` ?
```Java
 public void commit(List<MappedByteBuffer> buffers) {
    final Trace trace = new Trace(buffers);
    for (ExtOperation operation : this.operations) {
      operation.setup();
      this.traceExtOperation(operation, trace);
    }
  }
```